app/js/Icon.js

```diff
@@ -22,6 +23,7 @@ class Icon {
         this.iconPath.fillColor = null;
         this.iconBase = iconBase;
         this.iconShadow = new IconShadow(iconPath, iconBase);
+        this.banner = new Banner(iconPath, iconBase);
```

Comment:
Why is the banner part of the Icon class? Should those be logically separated, much like IconBase and Icon are?